startup/11-temperature-controller.py

```diff
@@ -257,9 +257,14 @@ class LinkamFurnace(PVPositioner):
     stop_signal = C(EpicsSignal, 'RAMP:CTRL:SET')
     temperature = C(EpicsSignal, "TEMP")
 
-    def set(self, *args, timeout=None, **kwargs):
-        
-        return super().set(*args, timeout=timeout, **kwargs)
+    def set(self, new_position, *args, timeout=None, **kwargs):
+
+        # Check if new setpoint is within one degree of current temperature, and if the controller is
+        # at it's setpoint. If it is, return that the move was successful instantly to avoid lockup.
+        if abs(self.readback.get() - new_position) < 1 and self.status.get() & 4 == 1:
```

Comment:
Will we ever want to parametrize `1`s on this line? Maybe we can have them as the object's attribute.

Reply:
That's probably a good idea. Initially I had them set to exactly equal, but remembered that the device will reach "setpoint" usually slightly before or after the actual target setpoint.